samples/snippets/src/main/java/com/example/analytics/QuickstartSample.java

```diff
@@ -16,82 +16,104 @@
-import com.google.analytics.data.v1alpha.DateRange;
-import com.google.analytics.data.v1alpha.Dimension;
-import com.google.analytics.data.v1alpha.Entity;
-import com.google.analytics.data.v1alpha.Metric;
-import com.google.analytics.data.v1alpha.Row;
-import com.google.analytics.data.v1alpha.RunReportRequest;
-import com.google.analytics.data.v1alpha.RunReportResponse;
+import com.google.analytics.data.v1beta.RunReportRequest;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import java.io.FileInputStream;
 
 public class QuickstartSample {
 
   // This is an example snippet that calls the Google Analytics Data API and runs a simple report
   // on the provided GA4 property id.
-  static void sampleRunReport(String ga4PropertyId) throws Exception {
+  static void sampleRunReport(String propertyId, String credentialsJsonPath) throws Exception {
+    /**
+     * TODO(developer): Uncomment this variable and replace with your
+     * Google Analytics 4 property ID before running the sample.
+     */
+    // propertyId = "YOUR-GA4-PROPERTY-ID";
+
+    // [START google_analytics_data_initialize]
+    /** TODO(developer): Uncomment this variable and replace with a valid path to
+     * the credentials.json file for your service account downloaded from the
+     * Cloud Console.
+     * Otherwise, default service account credentials will be derived from
+     * the GOOGLE_APPLICATION_CREDENTIALS environment variable.
+     */
+    // credentialsJsonPath = "/path/to/credentials.json";
+
     // Instantiates a client using default credentials.
     // See https://cloud.google.com/docs/authentication/production for more information
     // about managing credentials.
-    try (AlphaAnalyticsDataClient analyticsData = AlphaAnalyticsDataClient.create()) {
-      RunReportRequest request = RunReportRequest.newBuilder()
-          .setEntity(Entity.newBuilder().setPropertyId(ga4PropertyId))
-          .addDimensions(
-              Dimension.newBuilder().setName("city"))
-          .addMetrics(Metric.newBuilder().setName("activeUsers"))
-          .addDateRanges(
-              DateRange.newBuilder().setStartDate("2020-03-31").setEndDate("today")).build();
-
-      // Make the request
-      RunReportResponse response = analyticsData.runReport(request);
-
-      System.out.println("Report result:");
-      for (Row row : response.getRowsList()) {
+    BetaAnalyticsDataClient analyticsData;
+    if (credentialsJsonPath.isEmpty()) {
+      // Using a default constructor instructs the client to use the credentials
+      // specified in GOOGLE_APPLICATION_CREDENTIALS environment variable.
+      analyticsData = BetaAnalyticsDataClient.create();
+    } else {
+      // Explicitly use service account credentials by specifying
+      // the private key file.
+      GoogleCredentials credentials = GoogleCredentials
+          .fromStream(new FileInputStream(credentialsJsonPath));
```

Comment:
Wow, that page is missing a bit.  The authoritative pages are:
* https://google.aip.dev/auth/4110
* https://google.aip.dev/auth/4115
* https://google.aip.dev/auth

Reply:
b/181376135 filed.